### ISSUE TITLE
[BUG] Package breaks `npm ci` command

### ISSUE THREAD

**Issue by Twometer** [Soontao/light-odata#586]
## Describe the bug

After installing the `@odata/client` package, `npm ci` stops working.

## System Environment

* OS: Windows 11
* Node Version: 18.12.0
* This Library Version: 2.21.1

## To Reproduce

1. Create empty npm project
2. Run `npm install @odata/client`
3. Run `npm ci`

## Expected behavior
The command finishes without error

## Additional context
Here is the error I get, for reference
```
$ npm ci

npm ERR! code EBADPLATFORM
npm ERR! notsup Unsupported platform for @esbuild/android-arm@0.15.13: wanted {"os":"android","arch":"arm"} (current: {"os":"win32","arch":"x64"})
npm ERR! notsup Valid OS:    android
npm ERR! notsup Valid Arch:  arm
npm ERR! notsup Actual OS:   win32
npm ERR! notsup Actual Arch: x64
```

**Comment by Soontao** [comment]
interesting, windows on arm, let me check with esbuild

**Comment by Soontao** [comment]
Hi @Twometer 

I released a new version `2.21.2`, please have a try. 
I think I realized the nature of this issue, but its hard to explain, its similar to the well-known fsevents issue I think, and caused by npm/package-lock and optional dependencies.

Best Regards,
Theo

**Comment by Twometer** [comment]
Hi @Soontao,

I now get a different error:
```
npm ERR! code EBADPLATFORM
npm ERR! notsup Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin"} (current: {"os":"win32","arch":"x64"})
npm ERR! notsup Valid OS:    darwin
npm ERR! notsup Valid Arch:  undefined
npm ERR! notsup Actual OS:   win32
npm ERR! notsup Actual Arch: x64
```

**Comment by Soontao** [comment]
Hi @Twometer 

I'm so sorry just see your update today, really sorry for this, I've released a new version `2.21.3` to fix the `fsevents` issue, could you have a try ?

Best Regards,
Theo

**Comment by Twometer** [comment]
Hi @Soontao,

Thank you, the bug seems to be fixed now :)
